memoapp.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+# myapp.rb
+require 'bundler/setup'
+require 'json'
+require 'rubygems'
+require 'sinatra'
+require 'sinatra/reloader'
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+get '/' do
+  @memos = load_data['memos']
+  erb :top
+end
+
+get '/memo/compose' do
+  erb :newmemo
+end
+
+post '/memo/compose' do
+  data = load_data
+  id = calc_maxid(data)
+  newmemo = { 'id' => id, 'title' => params[:title], 'memo' => params[:memo] }
+  data['memos'].push(newmemo)
+  write_data(data)
+  redirect to "/memo/#{id}"
+end
+
+get '/memo/:id' do
+  id = params[:id]
+  @memo = load_memo(id)
+  erb :showmemo
+end
+
+get '/memo/:id/edit' do
+  id = params[:id]
+  @memo = load_memo(id)
+  erb :editmemo
+end
+
+patch '/memo/:id' do
+  id = params[:id].to_i
+  rewrit_memo = { 'id' => id, 'title' => params[:title], 'memo' => params[:memo] }
```

Comment:
```suggestion
  rewrite_memo = { 'id' => id, 'title' => params[:title], 'memo' => params[:memo] }
```

eが抜けてますー。

ただ動詞で始めるとメソッドっぽく見えてしまうので、変数名を見直した方がいいかもしれません。

[モデルやメソッドに名前を付けるときは英語の品詞に気をつけよう \- Qiita](https://qiita.com/jnchito/items/459d58ba652bf4763820)